src/App.tsx

```diff
@@ -1,14 +1,50 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { User } from './types/User';
+
+export const App: React.FC = React.memo(() => {
+  const [todos, setTodos] = useState< {
+    todos: Todo[]
+    originalTodos: Todo[];
+    searchQueryTodo: Todo[];
+  }>({
+    todos: [],
+    originalTodos: [],
+    searchQueryTodo: [],
+  });
```

Comment:
it's better not to use one state for these 3 variables. you can save only all original todos (coming from API). then you can create a filter and search query states and use `useMemo` hook for filtering original todos. it will be better solution.